torch/csrc/distributed/autograd/functions/dist_accumulate_grad.cpp

```diff
@@ -0,0 +1,37 @@
+#include <torch/csrc/distributed/autograd/functions/dist_accumulate_grad.h>
+
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+#include <torch/csrc/autograd/functions/utils.h>
+#include <torch/csrc/autograd/variable.h>
+
+namespace torch::distributed::autograd {
+
+DistAccumulateGrad::DistAccumulateGrad(
+      std::shared_ptr<AccumulateGrad> accumulateGrad,
```

Comment:
> [performance-unnecessary-value-param] warning: the parameter 'accumulateGrad' is copied for each invocation but only used as a const reference; consider making it a const reference

Don't quite understand this suggestion. The safe way to use `shared_ptr` is copying the value so that the underlying instance never gets freed.